fiasko_bro/code_validator.py

```diff
@@ -121,29 +126,36 @@ def _run_warning_validators_until(self, failed_error_group_name, arguments):
             )
         return warnings
 
-    def validate(self, repo_path, original_repo_path=None, check_repo_size=True, **kwargs):
-        self.validator_arguments.update(kwargs)
-        self.validator_arguments['whitelists'] = self.whitelists
-        self.validator_arguments['blacklists'] = self.blacklists
-        max_num_of_py_files = self.validator_arguments['max_num_of_py_files']
-        if check_repo_size:
-            if is_repo_too_large(repo_path, max_num_of_py_files, original_repo_path):
-                return [('repo is too large', '')]
-        self.validator_arguments['solution_repo'] = LocalRepositoryInfo(
-            repo_path)
-        if original_repo_path:
-            self.validator_arguments['original_repo'] = LocalRepositoryInfo(
-                original_repo_path)
+    def run_validator_group(self, group, pre_validation=False, *args, **kwargs):
```

Comment:
Let's rename `pre_validation` to `needs_to_run_warnings` to be more explicit about what that parameter actually does.